uaclient/api/u/pro/attach/auto/full_auto_attach/v1.py

```diff
@@ -0,0 +1,168 @@
+import time
+from typing import List, Optional
+
+from uaclient import actions, cli, contract, entitlements, exceptions, messages
+from uaclient.api.api import APIEndpoint
+from uaclient.api.data_types import AdditionalInfo
+from uaclient.api.errors import APIError
+from uaclient.clouds import AutoAttachCloudInstance  # noqa: F401
+from uaclient.config import UAConfig
+from uaclient.data_types import (
+    DataObject,
+    Field,
+    IntDataValue,
+    StringDataValue,
+    data_list,
+)
+from uaclient.entitlements.entitlement_status import CanEnableFailure
+
+
+class BetaEntitlementFoundError(APIError):
+    pass
+
+
+class EntitlementNotEnabledError(APIError):
+    pass
+
+
+class FullAutoAttachFailureError(APIError):
+    pass
```

Comment:
As we discussed today, let us not create specific `APIError` exceptions here, instead just rely on the `message_code` we already have for the existing exceptions. However, let's double check to see if the exception being raise are lacking `message_codes`. If they are, we need to add them to the exceptions